cardinal/message/manager.go

```diff
@@ -24,20 +25,20 @@ func NewManager() *Manager {
 }
 
 func (m *Manager) RegisterMessage(msgType types.Message, msgReflectType reflect.Type) error {
-	name := msgType.Name()
```

Comment:
What I think we should do here is make msgType.ID() be set to what you are currently defining as msgType.FullName(). Since FullName() is guaranteed to be unique, it's a valid substitute for ID() and is actually better we can also remove ID() altogether, therefore resolving the TODO I left below on msgType.SetID().

Reply:
eh, would prefer to make the removal of ID a linear ticket. that has some cascading changes across tests and would prefer to keep this PR focused